### DISCUSSION TITLE
Tailwind 3 is not rerendering when using Storybook

### DISCUSSION THREAD

**Discussion by Sannnji** [tailwindlabs/tailwindcss#9020] [Help] [answered]
repo: https://github.com/Sannnji/storybook-issue-repro

I'm currently using Tailwind with Storybook for a project and the story components are not rerendering after changing the properties 

![Screenshot 2022-08-03 at 16-30-50 Button - Primary ⋅ Storybook](https://user-images.githubusercontent.com/57771801/182706088-e974d7a8-0692-4211-befc-5b158a3ee823.png)

The only Tailwind classes that will rerender are the ones that are used in our projects

For Example: 
My button component has a `size` prop and `borderRadius` prop. Both initial values like `h-10 w-56` and `rounded` will be rendered, but when you modify the class value in the storybook control it doesn't rerender the change. It just invalidates the class. But if I have the button component referenced somewhere else in my project with different props like `h-20 w-80` and `rounded-xl`, then those specific class values will work in storybook control. (My repo has the example set up)

I've tried using this solution [postcss-loader, adding stories directory to my tailwind config](https://github.com/tailwindlabs/tailwindcss/discussions/6570) and explicitly setting purge to false, but nothing works. 

From my pov it seems like tailwind is still purging all the unused classes when using storybook. No classes work unless they are explicitly used within the project.

**Comment by wongjn** [comment, marked as answer]
You are using Tailwind v3 and in v3, no purging happens. In actuality, it is the opposite of Tailwind v2 (non-JIT) – in v2, Tailwind generates the classes and then scans your code and purges unused classes. In v3, it scans your code first and then generates the classes from what it finds.

So in how this relates to your issue, `h-20`,  `w-80` and `rounded-xl` work because they exist in your codebase when the Tailwind content scan happens (at build time) and thus these classes are generated, they exist and are available for use. Other classes you may try don't work because they don't exist because the Tailwind content scan never saw them and thus never generated them. If you need more classes that won't exist in your codebase, you should look at [safelisting them](https://tailwindcss.com/docs/content-configuration#safelisting-classes).

**Reply by Sannnji** [reply]
Thank you so much for that. I thought I had incorrectly implemented tailwind and storybook.
